lib/spack/spack/cmd/uninstall.py

```diff
@@ -147,46 +175,71 @@ def num_installed_deps(pkg):
     for item in packages:
         item.do_uninstall(force=force)
 
+    # write any changes made to the active environment
+    if ev.active:
+        ev.active.write()
 
-def get_uninstall_list(args):
-    specs = [any]
-    if args.packages:
-        specs = spack.cmd.parse_specs(args.packages)
 
+def get_uninstall_list(args, specs):
     # Gets the list of installed specs that match the ones give via cli
     # takes care of '-a' is given in the cli
     uninstall_list = find_matching_specs(specs, args.all, args.force)
 
-    # Takes care of '-d'
-    dependent_list = installed_dependents(uninstall_list)
-
-    # Process dependent_list and update uninstall_list
-    has_error = False
-    if dependent_list and not args.dependents and not args.force:
-        for spec, lst in dependent_list.items():
-            tty.error("Will not uninstall %s" % spec.cformat("$_$@$%@$/"))
-            print('')
-            print('The following packages depend on it:')
-            spack.cmd.display_specs(lst, **display_args)
-            print('')
-            has_error = True
+    # Takes care of '-R'
+    spec_dependents = installed_dependents(uninstall_list)
+    spec_envs = dependent_environments(uninstall_list)
+
+    # Process spec_dependents and update uninstall_list
+    has_error = not args.force and (
+        (spec_dependents and not args.dependents) or
+        spec_envs)
```

Comment:
(documentation) It looks like you have to use `--force` if the same spec is installed in more than one environment - is that true? Not necessarily a problem, but IMO worth documenting.

Reply:
This one is a stopgap -- once we implement garbage collection, uninstall should just remove.

Reply:
Added to the argparse documentation of `--force`